### PR TITLE
Fix bugs in category assigment/ordering and Main Menu module having no menu assigments

### LINKED ISSUE #23: Module menu assigments are broken
Latest fixes seem to break module assignments when they were attached only to some menu items. Result is that every assignment that wasn't 'All' becomes 'None'. So the menu mapping code is still needed -- if only for Itemid=1 (and rest being x => x).

### LINKED ISSUE #29: Uncategorized items (no category) get moved into wrong category
This bug is at least in articles, but most likely apply to every other component that has categories, too.

If Joomla 1.5 site had articles that were not assigned to any category, they get moved to the first valid category, which changes how the site behaves.

```
Title:'My Article' Section: '', Category: ''
```

becomes:

```
Title:'My Article' Category: 'My Products'
```

where it should be:

```
Title:'My Article' Category: 'Uncategorized'
```

### PR DESCRIPTION
**Pull request by mahagr**
- Fix #23 Main Menu module looses its menu assigments
- Fix category ordering to be the same as in Joomla 1.5
- Fix #29 Uncategorized items (no category) get moved into wrong category

### CODE REVIEW

File: trunk/admin/includes/migrate_categories.php [hunk 1]
```diff
@@ -115,12 +115,12 @@ protected function setDestinationData()
 
 			$category['asset_id'] = null;
 			$category['parent_id'] = isset($catmap[$category['extension']]->new) ? $catmap[$category['extension']]->new : 1;
-			$category['lft'] = null;
+			$category['lft'] = $i;
```
Comment (mahagr) [inline trunk/admin/includes/migrate_categories.php:117]: This is used in rebuild()

File: trunk/admin/includes/migrate_content.php [hunk 2]
```diff
@@ -45,15 +45,25 @@ protected function &getSourceData()
 
 		$rows = parent::getSourceData(
 			'`id`, `title`, `alias`, `title_alias`, `introtext`, `fulltext`, `state`, '
-		 .'`mask`, o.new AS catid, `created`, `created_by`, `created_by_alias`, '
-		 .'`modified`, `modified_by`, `checked_out`, `checked_out_time`, `publish_up`, `publish_down`, '
-		 .'`images`, `urls`, `attribs`, `version`, `parentid`, `ordering`, `metakey`, `metadesc`, '
-     .'`access`, `hits` ',
-		 'LEFT JOIN jupgrade_categories AS o ON o.old = c.catid',
```
Comment (mahagr) [inline trunk/admin/includes/migrate_content.php:52]: This JOIN caused articles to be lost if category mapping didn't exit.

File: trunk/admin/includes/migrate_modules.php [hunk 3]
```diff
@@ -210,7 +210,7 @@ class jUpgradeModulesMenu extends jUpgrade
 	protected function &getSourceData()
 	{
 		// Creating the query
-		$where = "m.moduleid NOT IN (1,2,3,4,8,13,14,15)";
```
Comment (mahagr) [inline trunk/admin/includes/migrate_modules.php:213]: Main Menu module = 1, what are the others?

**Comment by mahagr** [comment]
@fastslack Here's a pull that fixes a few issues. Can you also check if there are similar issues in other components that use categories?

**Comment by fastslack** [comment]
Thanks Matias

**Pull request merged**